lib/charms/layer/kubernetes_common.py

```diff
@@ -159,7 +159,7 @@ def calculate_and_store_resource_checksums(checksum_prefix, snap_resources):
         db.set(key, checksum)
 
 
-def get_ingress_address(endpoint_name):
+def get_ingress_address(endpoint_name, ignore_addresses=[]):
```

Comment:
Not exactly an issue in this case, but it's always a bad idea to use a mutable value (e.g., list) as a default param. This should work fine with the subsequent logic instead:

```suggestion
def get_ingress_address(endpoint_name, ignore_addresses=None):
```

Reply:
makes sense, I changed this in the new version of the patch.